pkg/dockerregistry/server/auth.go

```diff
@@ -535,3 +545,11 @@ func verifyPruneAccess(ctx context.Context, client client.SubjectAccessReviews)
 	}
 	return nil
 }
+
+func isMetricsBearerToken(ctx context.Context, token string) bool {
+	config := ConfigurationFrom(ctx)
+	if config.Metrics.Enabled {
+		return config.Metrics.Secret == token
+	}
```

Comment:
> If you are already doing auth via the master, we want metrics to use that behavior. Static secret is only for things that aren't integrated with master auth (which is a limited set, but registry is under)

We have already discussed [this](https://github.com/openshift/origin/pull/12711#issuecomment-276105926). Registry metrics are not part of the master API. The metrics belongs to the registry server only. You propose to make some virtual object (`registrymetrics` for example) and use SAR to check access to it ?

> Actually, does the registry support cert auth (is it using the correct filter for that)? If not, I'm ok with static secret.

The registry does not have a ready solution for it. Additionally, cert authentication will work only if HTTPS enabled. It means that it will be impossible to restrict access to metrics in an insecure registry.

@mfojtik @smarterclayton I propose to leave the possibility of shared secret authentication and add cert authentication later (in 3.7) because we don't have time to add it for 3.6. ok?